src/test/regress/expected/multi_select_distinct.out

```diff
@@ -0,0 +1,790 @@
+     15283 |     3
+     87761 |     3
+    136884 |     3
+    149926 |     3
+    160895 |     3
+    177771 |     3
+    188804 |     3
+    199146 |     3
+(11 rows)
+
+	
+-- explain the query to see actual plan
+EXPLAIN (COSTS FALSE)
+	SELECT DISTINCT l_partkey, count(*)
+		FROM lineitem_hash_part
+		GROUP BY 1 
+		HAVING count(*) > 2
+		ORDER BY 1;
+                                             QUERY PLAN                                             
+----------------------------------------------------------------------------------------------------
+ Sort
+   Sort Key: remote_scan.l_partkey
+   ->  HashAggregate
+         Group Key: remote_scan.l_partkey
+         Filter: (COALESCE((pg_catalog.sum(remote_scan.worker_column_3))::bigint, '0'::bigint) > 2)
+         ->  Custom Scan (Citus Real-Time)
+               Task Count: 4
+               Tasks Shown: One of 4
+               ->  Task
+                     Node: host=localhost port=57637 dbname=regression
+                     ->  HashAggregate
+                           Group Key: l_partkey
+                           ->  Seq Scan on lineitem_hash_part_360290 lineitem_hash_part
+(13 rows)
+
+-- distinct on non-partition column and avg
+SELECT DISTINCT l_partkey, avg(l_linenumber)
+	FROM lineitem_hash_part
+	WHERE l_partkey < 500
+	GROUP BY 1 
+	HAVING avg(l_linenumber) > 2
+	ORDER BY 1;
+ l_partkey |        avg         
+-----------+--------------------
+        18 | 7.0000000000000000
+        79 | 6.0000000000000000
+       149 | 4.5000000000000000
+       175 | 5.0000000000000000
+       179 | 6.0000000000000000
+       182 | 3.0000000000000000
+       222 | 4.0000000000000000
+       278 | 3.0000000000000000
+       299 | 7.0000000000000000
+       308 | 7.0000000000000000
+       309 | 5.0000000000000000
+       321 | 3.0000000000000000
+       337 | 6.0000000000000000
+       364 | 3.0000000000000000
+       403 | 4.0000000000000000
+(15 rows)
+
+-- distinct on multiple non-partition columns
+SELECT DISTINCT l_partkey, l_suppkey
+	FROM lineitem_hash_part
+	WHERE l_shipmode = 'AIR' AND l_orderkey < 100
+	ORDER BY 1, 2;
+ l_partkey | l_suppkey 
+-----------+-----------
+      2132 |      4633
+      4297 |      1798
+     37531 |        35
+     44161 |      6666
+     44706 |      4707
+     67831 |      5350
+     85811 |      8320
+     94368 |      6878
+    108338 |       849
+    108570 |      8571
+    137267 |      4807
+    137469 |      9983
+    173489 |      3490
+    196156 |      1195
+    197921 |       441
+(15 rows)
+
+	
+EXPLAIN (COSTS FALSE)
+	SELECT DISTINCT l_partkey, l_suppkey
+		FROM lineitem_hash_part
+		WHERE l_shipmode = 'AIR' AND l_orderkey < 100
+		ORDER BY 1, 2;
+                                             QUERY PLAN                                              
+-----------------------------------------------------------------------------------------------------
+ Sort
+   Sort Key: remote_scan.l_partkey, remote_scan.l_suppkey
+   ->  HashAggregate
```

Comment:
Just noting an observation. Here, PostgreSQL picks Sort + Unique, while we pick HashAggregate.